apollo-api/src/main/java/com/spotify/apollo/Response.java

```diff
@@ -40,10 +42,33 @@
 
   /**
    * The response headers.
+   * Deprecated in favor of {@link Response#headerEntries()}
    */
+  @Deprecated
   Map<String, String> headers();
 
   /**
+   * The response headers
+   */
+  Iterable<Map.Entry<String, String>> headerEntries();
```

Comment:
Iterator is generally wrong, since they are one-use only. Iterable is better in that case.

I would still prefer List; it's still quite generic, and I feel like we have committed to retaining order. We could even specify that in the javadocs.